### DISCUSSION TITLE
Pixel 6 Pro bonding issues

### DISCUSSION THREAD

**Discussion by cwpeterson** [Navid200/xDrip#120] [Dexcom]
Posted on the Facebook page and was told to go through the intermittent connectivity documentation, but that didn't help. Here's my issue:

Recently switched to Pixel 6 Pro and running into a situation where it gets stuck in a "Bonding" state. I have previously managed to get it working and have it be rock solid, until I rebooted yesterday for an update. Now it will connect once, then get stuck Bonding. It connects but gets no data. If I restart the collector it will get data then get stuck again. If I forget the device and re-pair the transmitter, I get one or two readings and back to stuck bonding. I've reviewed my settings and compared with the documentation and it matches up.

I am running the latest nightly from 11/18.

Logs say device is marked as bonded but we are being asked to unbond, but preference settings prevents it. (unbond)

Of course if I allow it to unbond I have to catch the pairing request every time.

It will sporadically connect and get data but is offline for 30-45 minutes or longer before it pulls data again.

Really hoping to get some answers on this. I love xDrip and rely it to wake me overnight when I get too low.

**Comment by Navid200** [comment]
What do you mean by "Bonding state" or "stuck bonding"?
Are these references to some guides or documentation?  Can you provide links?

Have you disabled Bluetooth battery optimization?

**Reply by cwpeterson** [reply]
Yes I have disabled bluetooth battery optimization.

I've uploaded a screenshot of the logs showing the bonding message and one of the status screen.

![Screenshot_20211117-155332 (1)](https://user-images.githubusercontent.com/12939749/142459931-8bcea86d-689d-478d-a35f-4c120f08888a.png)
![Screenshot_20211118-104957](https://user-images.githubusercontent.com/12939749/142459947-857d79cb-6305-41a3-a7ed-5ba34291d5b9.png)

**Comment by Navid200** [comment]
Can you also show a screenshot of the Android Bluetooth settings page?
It should look something like this:
https://github.com/Navid200/xDrip/blob/master/Documentation/images/BT_Opt.png

**Reply by cwpeterson** [reply]
Yes see attached. I went in and reset this again and have attempted to repair. Will monitor.

![Screenshot_20211118-105818](https://user-images.githubusercontent.com/12939749/142461348-f274225d-5583-48f4-bcbc-5b144153f826.png)

**Reply by cwpeterson** [reply]
Nope. Back to the same. Got data, now stuck in "bonding" state.

**Comment by Navid200** [comment]
Would you please take longshots (single-screen images) of the following three pages and post?
Settings -> G5/G6 Debug Settings
Settings -> Less common settings -> Bluetooth Settings
Settings -> Less common settings -> Other misc. options

**Reply by cwpeterson** [reply]
Updated screenshots.
![Screenshot_20211118-114103](https://user-images.githubusercontent.com/12939749/142468526-491fb57f-0b54-4db1-a4f3-e8424c0c191a.png)
![Screenshot_20211118-114116](https://user-images.githubusercontent.com/12939749/142468537-6527753c-8448-42ae-b09b-e7a18c36372f.png)
![Screenshot_20211118-114127](https://user-images.githubusercontent.com/12939749/142468545-115e08f6-5482-4283-a4c9-002d29bf592c.png)

**Reply by Navid200** [reply]
Please edit that post, delete the duplicate image, and add the G5/G6 Debug missing page.

**Reply by cwpeterson** [reply]
I took new screenshots and cleaned up the old ones.

**Comment by Navid200** [comment]
If you enable minimize scanning and avoid scanning, and enable "Allow OB1 unbonding", how often will you need to approve a pair request?

**Reply by cwpeterson** [reply]
I disabled the scanning options since this issue started per the documentation. If I do the allow unbonding it prompts to pair every 5 minutes.

**Comment by Navid200** [comment]
Can you also show this page?
https://github.com/Navid200/xDrip/blob/master/Documentation/images/location-android11.png

**Reply by cwpeterson** [reply]
Sure, location permission is definitely on

![Screenshot_20211118-123330](https://user-images.githubusercontent.com/12939749/142475900-8f9ac9ec-b965-4787-9606-badfc209d95a.png)
.

**Comment by Navid200** [comment]
Would you like to enable this developer option as an experiment to see if it may improve anything?

![Screenshot_20211118-134049](https://user-images.githubusercontent.com/51497406/142477346-46e7e4d9-ec08-470e-a946-2aed19319d6c.png)

**Reply by cwpeterson** [reply]
Ok, where is that setting? I'm not seeing developer options.

**Reply by Navid200** [reply]
It's in Android settings.  You need to go to Android settings -> About phone, and tap on Build number multiple times.  After you tap on it a few times, it tells you how many more times you need to tap.  After it enables Developer options, you leave and find developer options under Android Settings -> System -> Developer options

**Reply by cwpeterson** [reply]
Oh I thought you were referring to an xDrip setting. I can take a look at enabling that. Then how do you want me to proceed?

**Reply by Navid200** [reply]
Let us know if it makes the disconnects more or less.

**Reply by cwpeterson** [reply]
Unfortunately no change.

![Screenshot_20211118-134010](https://user-images.githubusercontent.com/12939749/142485504-da23e95b-457e-4ef8-9520-da145e5fa7a4.png)
![Screenshot_20211118-134255](https://user-images.githubusercontent.com/12939749/142485881-f4f02acc-069e-4a34-babc-6e8988c187ef.png)

**Comment by Navid200** [comment]
This is perfectly fine:
![142485881-f4f02acc-069e-4a34-babc-6e8988c187ef](https://user-images.githubusercontent.com/51497406/142486967-770104f5-491a-44fe-b39c-033523f87195.png)

**Reply by cwpeterson** [reply]
No it's not. It connects but gets NO data. That's the problem.

**Reply by Navid200** [reply]
I understand.  
But, you are showing a system status page that is perfect.  It helps to be sure that you are not worried about Brain state deep sleeping.

Please always pay attention to last connected.
Does it always show a value less than 5 minutes when the signal loss occurs?

**Reply by cwpeterson** [reply]
It connects yes, every 5 minutes. But not getting any readings unless I forget the device and re-pair. That is the issue. It says bonded but no readings. Event log says it's bonded but wants to unbond, which would force another pair message if I let it unpair automatically. I've never had this much trouble with getting it working again. I don't know if it's something in the most recent patch that was applied. It was finicky when I switched to the new phone, but I was always able to eventually get it going, now it simply won't give consistent readings. See new screenshots. Same issue, time is further between readings. It may eventually get a reading, but 45 minutes, an hour? Who knows? It's not consistently working every 5 minutes as it should and has previously.
![Screenshot_20211118-141258](https://user-images.githubusercontent.com/12939749/142489830-f41411f8-37b6-409c-9984-925b8a1719f0.png)
![Screenshot_20211118-141309](https://user-images.githubusercontent.com/12939749/142489838-b66272cd-6233-4173-8086-f8307ae4ab32.png)

**Reply by cwpeterson** [reply]
Anyone have any other ideas? It gets readings randomly but no consistency. 30 minutes, an hour. Last time was over 70 minutes ago. I tried a reboot and no luck. Had to remove device again and manually pair. It got data 2 times in a row and now nothing in the past 15 minutes.

**Reply by Navid200** [reply]
> Had to remove device again and manually pair. 

What do you mean you had to "remove" device?

**Reply by cwpeterson** [reply]
Sorry forget device. On the status screen.

**Comment by Navid200** [comment]
> Recently switched to Pixel 6 Pro 

Have you disabled collection on the previous phone?  Or, have you uninstalled xDrip from the previous phone?

**Reply by cwpeterson** [reply]
Yes. I've been on the Pixel 6 Pro since launch, 2 weeks now? Old phone uninstalled, wiped, shut down. What else can we try? I had a bad low overnight and unfortunately xDrip had missed readings for over 40 minutes at that point. Fortunately I heard my pump alarm and woke up. Getting a bit desperate to get this going again. I love xDrip and really rely on it daily.

**Reply by Navid200** [reply]
What pump?

**Reply by cwpeterson** [reply]
Tandem tSlim. I have no connectivity issues with it and the sensor or with my phone. Have used them side by side with xDrip for a couple years.

**Reply by cwpeterson** [reply]
I've no explanation why but in the past 32 hours it's not missed a single reading and not a single message about unbonding. So strange.
